### PR TITLE
homework1

### PR DESCRIPTION
**Pull request by khushikdu**
homework1

### CODE REVIEW

**Comment by khushikdu** [comment]
| Metric | Value |
| ------ | ----- |
| alert_status | OK |
| bugs | 1 |
| code_smells | 25 |
| reliability_rating | 2.0 |
| security_rating | 1.0 |
| vulnerabilities | 0 |

File: backend/homework/1/logging/src/main/java/org/example/Main.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class Logback {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Logback.class);
+
+    public static void main(String[] args) {
+        LOGGER.info("This is an INFO level log message!");
```
Comment (Sagun15) [inline backend/homework/1/logging/src/main/java/org/example/Main.java:11]: Why it is added?

File: backend/homework/1/data/src/main/java/org/example/Main.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            for (int i = 0; i < s.length; i++) {
+                if (s[i].getId() == id) {
+                    return s[i];
+                }
+            }
+            return null;
+        }
+        catch(Exception e){
+            Logback.getMsg().debug("Terminating program");
+            System.out.println("Id not found");
```
Comment (Sagun15) [inline backend/homework/1/data/src/main/java/org/example/Main.java:31]: How do you know the exception is due to id not found? Use generic messages

File: backend/homework/1/core/src/main/java/org/example/Main.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        this.grade=grade;
+    }
+
+    public  int getId(){
+        return id;
+    }
+    public  String getName(){
+        return name;
+    }
+    public  int getAge(){ return age;    }
```
Comment (Sagun15) [inline backend/homework/1/core/src/main/java/org/example/Main.java:25]: Use multi-line statements

File: backend/homework/1/app/src/main/java/org/example/Main.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        int i=0;
+        //System.out.print("Enter the number of students : ")
+        while(true) {
+            System.out.print("\n1. Add\n2. Update\n3. Retrieve\n4. Exit\nEnter your choice : ");
+            int choice=sc.nextInt();
+            switch (choice) {
+                case 1:
+                    if(i<10) {
+                        s[i] = r.addStudent(s[i]);
+                        //r.display(s[i]);
```
Comment (Sagun15) [inline backend/homework/1/app/src/main/java/org/example/Main.java:23]: Why this comment is added? If not required, please remove it

File: backend/homework/1/app/src/main/java/org/example/Main.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    {
+        Scanner sc=new Scanner(System.in);
+        StudentRepository r=new StudentRepository();
+
+        Student []s=new Student[10];
+
+        int i=0;
+        //System.out.print("Enter the number of students : ")
+        while(true) {
+            System.out.print("\n1. Add\n2. Update\n3. Retrieve\n4. Exit\nEnter your choice : ");
```
Comment (Sagun15) [inline backend/homework/1/app/src/main/java/org/example/Main.java:17]: Use loggers

File: backend/homework/1/app/src/main/java/org/example/Main.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+public class Main
+{
+    public static void main(String[] args)
+    {
+        Scanner sc=new Scanner(System.in);
+        StudentRepository r=new StudentRepository();
+
+        Student []s=new Student[10];
+
+        int i=0;
```
Comment (Sagun15) [inline backend/homework/1/app/src/main/java/org/example/Main.java:14]: Use meaningful variable names

File: backend/homework/1/app/src/main/java/org/example/Main.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+import org.example.StudentRepository;
+import java.util.Scanner;
+public class Main
+{
+    public static void main(String[] args)
+    {
+        Scanner sc=new Scanner(System.in);
+        StudentRepository r=new StudentRepository();
+
+        Student []s=new Student[10];
```
Comment (Sagun15) [inline backend/homework/1/app/src/main/java/org/example/Main.java:12]: Why size is hardcoded? Also use meaningful variable names

**Review by Sagun15 — COMMENTED** [review]
Feel free to reach out if you have any doubts

**Comment by Sagun15** [comment]
Please update your post-comment.js file. Check kdu-backend channel.

**Comment by khushikdu** [comment]
| Metric | Value |
| ------ | ----- |
| alert_status | <b>:x: ERROR</b> |
| bugs | 1 |
| code_smells | 26 |
| reliability_rating | B (at least 1 Minor Bug)<br/>Note: A being best, E being worst |
| security_rating | A (0 Vulnerabilities)<br/>Note: A being best, E being worst |
| vulnerabilities | 0 |

**Pull request closed without merging**